### PR TITLE
Move typed clients into clientset folder

### PR DESCRIPTION
**Pull request by caesarxuchao**
Move typed clients from `pkg/client/typed/` to `pkg/client/clientset_generated/${clientset_name}/typed`.

The first commit changes the client-gen, the last commit updates the doc, other commits are just moving things around.

@lavalamp @krousey

### CODE REVIEW

File: pkg/client/clientset_generated/release_1_2/typed/core/v1/fake/doc.go [hunk 1]
```diff
@@ -14,5 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// This file is generated by client-gen with arguments: --clientset-name=release_1_2 --input=[api/v1,extensions/v1beta1]
```
Comment (caesarxuchao) [inline pkg/client/clientset_generated/release_1_2/typed/core/v1/fake/doc.go]: We should move this auto-generated comment to doc.go, but that will be another PR.

**Comment by krousey** [comment]
@lavalamp What are we promising with backwards compatibility. Do we only promise to keep the top-level clientset in the same place?

**Comment by caesarxuchao** [comment]
Probably we should keep a copy of the old typed clients. I can remove the commit "remove the old typed clients", but will wait for @lavalamp's opinion.

**Comment by krousey** [comment]
This looks good to me so long as @lavalamp signs off on the removal of the old packages being within our backwards compatibility promise.

**Comment by caesarxuchao** [comment]
@k8s-bot e2e test this, issue: #23321

**Comment by caesarxuchao** [comment]
Ping @lavalamp. This PR moves the typed clients. For backward compatibility, how about we duplicate them at the old directory?

**Comment by lavalamp** [comment]
If it's easy, that sounds like a good idea.

**Comment by caesarxuchao** [comment]
I add back the old typed clients. @krousey PTAL. Thanks!

**Comment by caesarxuchao** [comment]
Jenkins GCE Node e2e test failure looks like #23309

https://doc-0rqg0-0vpii-s-googleusercontent.commondatastorage.googleapis.com/gs/60vds4sgt4006psshqkrfipt5tek5nig/15v4jn41kl31oqmrvf3sqe6hlg6cplpj/1459485000000/kubernetes-jenkins/10785533137159089557/ChJrdWJlcm5ldGVzLWplbmtpbnMSPXByLWxvZ3MvcHVsbC8yMzYwOC9ub2RlLXB1bGwtYnVpbGQtZTJlLXRlc3QvMTczL2J1aWxkLWxvZy50eHQ?a=AGjQbs4KAKTjAtf9zSF6hL9X8T9uiDe6B2cp0zth0YZ7YPd_7f3dGPON4RMix60QReG-ePU-Q8AU6hfKyTUieYwIiIT_YgViWMoN0sw3UXG88O_ChwPBk4E7humVgdFbI-T_liaH0wFyeaIA0Krb0vGwfpy7aHsRABmzDnb1GD5n4eCTIgvdjuKTXfDZ-h5sD4Ss3hWg2TKkBZLREepErXQOsw4q-OyxpvCYU69paeeWGHEhou_-5dFkvGfcOQT2CKcY1OkkOGnbOKtIVXkhJCPKNmf_FELxxABEoQRH2KX4gjmX9_gln0YmdobagUkazKEH3VD0-fTZHL3Hdq_RHYRVL7ONJZauCyvHQLGh3BrXAv-YuFd0B3CzgmlfN1f0FJXfWtovGhPrY8LV-rDwv-T-9E_pQD5yEHf8xNSk_ABzEs8poK9yc__laA0s6upOg2r26IQhqbjU_xpPLsU3s2D4YJIdMSGygRyX_BWUrGzNR5aBVWQ9lpVQgoSIRVuvOxDES_l0LdqPqooMdAKkhoyD7of__kV9H2tYHBnguO2gY9j2-EeRUrON69Ryh-lBmAxZi6Rv5DQsyEoKWje-_oAcSuMrS4wwgfHATYYu1oKtGr9tdlfPiOwIWMB5XKREUJ3os5BT8-p3rifPMkShktMQS7NMt7sq6LAJdqbl2QiD5_iCbDHcHHvx0xSkSjg6IweT6XYrdECdHS6CL0KOcWCeIKi6W5FugUY71G73Y5Zw8JAXzp9MDGmEk5XLS1wkkb8gx2XknDT9

**Comment by caesarxuchao** [comment]
@k8s-bot GCE Node e2e test this, issue #23309

Don't know what's the magic word to trigger GCE Node e2e test.

**Comment by caesarxuchao** [comment]
@k8s-bot test node e2e

**Comment by caesarxuchao** [comment]
@k8s-bot test node e2e experimental

**Comment by krousey** [comment]
Sorry. Missed the latest update. Squash and LGTM

**Comment by caesarxuchao** [comment]
Thanks. I'm not squashing the PR because I'd like to keep the actual code change and generated code in two commits.
